docs/process-for-changing-ion-schema.md

```diff
@@ -0,0 +1,173 @@
+
+Once the new version of the spec has been merged (via one or more pull requests), Ion Schema libraries can be updated to add support for the features that were included in the new version.
+
+As before, announcements will be added to the Ion home and news pages.
+
+# Internal steps for considering a proposal
+
+*This section is addressed to the Ion Schema ** **maintainers**.* *It is included here in order to be transparent about the approval process.*
+
+## Requesting an RFC
+
+When a community member creates a feature request in `ion-schema`, we (the Ion Schema **maintainers**) will engage with the requester on the GitHub issue. The **maintainers** will come to a consensus on the proposal; if its value proposition is compelling, we will request an RFC to explore the technical details of its implementation in depth. If the proposal does not add value or goes against the design principles of Ion Schema, we will politely decline. However, this does not preclude the **feature requestor** or any other **contributor** from revising and resubmitting their proposal with new ideas or information.
+
+Either way, the Ion schema **maintainers** must respond with a justification for our decision. For a feature that is approved, a simple message about how the feature will be useful, aligns with the design goals, etc. will suffice. For a rejection, the justification should specifically explain why the idea does not add value to Ion Schema or how it goes against the design principles of Ion Schema.
+
+If we are interested in an idea but **feature requestor** does not want to create an RFC, we may decide to take it up ourselves, but we are not required to do so.
+
+## Approving an RFC
+
+The Ion Schema **maintainers** are responsible for implementing and maintaining both the spec and most of its implementations, so we are the biggest stakeholder. The RFC cannot move ahead without the **maintainers**’ agreement.
+We believe that it should be easy to iterate on Ion Schema, and so the most critical feedback that we can receive is whether something we are introducing is a misfeature or could otherwise cause problems later on. Feedback about the omission of desired features (or part of a feature) will also be considered, but we may decide to defer solving it if there are no backwards compatibility issues that the proposed feature would introduce.
+
+After feedback has been gathered from Ion Schema users and subject-matter experts, the Ion Schema **maintainers** will incorporate the feedback and make the final decision on the RFC.
+
+The outcome of this process will be shared on the `ion-schema` GitHub issue, resulting in the PR being merged or closed as appropriate.
+
+## Creating an RFC for a new Ion Schema version
```

Comment:
This section has a lot of good details on when to create a new schema version. I had a suggestion to add a consideration in this section, feel free to skip it if it doesn't make sense to add here. Should this also add a consideration of when to have major vs minor version based on current requests? For example, if the current feature request includes a breaking change then we might want to wait for a couple of more features to be added for a new major version release.

Reply:
No, I don't think we should delay a breaking change in order to batch them up (unless there is another breaking change that is _currently_ in flight). We've demonstrated that it is possible to have ISL 1.0 and ISL 2.0 import types from each other and mostly share an internal model in the implementations. The different ISL versions are really just feature gates for different syntax.